packages/patternfly-3/patternfly-react/src/components/DateTimePicker/TimePicker.js

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormControl, InputGroup, Icon, OverlayTrigger, Popover } from '../../';
+import TimeInput from './TimeComponents/TimeInput';
+
+class TimePicker extends React.Component {
+  getDateFromTime = time => {
+    if (Date.parse(time)) {
+      return new Date(time);
+    }
+    return new Date(`1/1/1 ${time}`);
+  };
+
+  formatDate = time => {
+    const { locale } = this.props;
+    const options = { hour: '2-digit', minute: '2-digit' };
+    if (Date.parse(time)) {
+      const parsedValue = new Date(time);
+      return parsedValue.toLocaleString(locale, options);
+    } else if (Date.parse(`1/1/1 ${time}`)) {
```

Comment:
what does it mean ?

Reply:
The date object in JavaScript requires a date to create a new object, and this function is used to we can use the Time Picker with time value only like this:
`<TimePicker value="2:22:31 PM" />`
Without this function the component will have to always get a date with the time:
`<TimePicker value="1/1/1 2:22:31 PM" />`